optimum/exporters/openvino/model_configs.py

```diff
@@ -2903,6 +2939,421 @@ def patch_model_for_export(
+                framework=framework,
+                dtype="bool",
+                max_value=2,
+            )
+        return super().generate(input_name, framework, int_dtype, float_dtype)
+
+    def get_vision_position_ids(self):
+        # Adopted from https://github.com/huggingface/transformers/blob/v4.51.3/src/transformers/models/phi4_multimodal/modeling_phi4_multimodal.py#L494-L512
+        import torch
+
+        batch_size = self.batch_size
+        max_im_h, max_im_w = self.height, self.width
+        max_nb_patches_h, max_nb_patches_w = max_im_h // self.patch_size, max_im_w // self.patch_size
+        boundaries = torch.arange(1 / self.num_patches_per_side, 1.0, 1 / self.num_patches_per_side)
+        position_ids = torch.full(
+            size=(
+                batch_size,
+                max_nb_patches_h * max_nb_patches_w,
+            ),
+            fill_value=0,
+        )
+        patch_attention_mask = torch.ones(
+            [self.batch_size, self.height // self.patch_size, self.width // self.patch_size], dtype=torch.int64
+        )
+        patch_attention_mask[0, self.height - 2 :] = 0
+
+        for batch_idx, p_attn_mask in enumerate(patch_attention_mask):
+            nb_patches_h = p_attn_mask[:, 0].sum()
+            nb_patches_w = p_attn_mask[0].sum()
+
+            fractional_coords_h = torch.arange(0, 1 - 1e-6, 1 / nb_patches_h)
+            fractional_coords_w = torch.arange(0, 1 - 1e-6, 1 / nb_patches_w)
+
+            bucket_coords_h = torch.bucketize(fractional_coords_h, boundaries, right=True)
+            bucket_coords_w = torch.bucketize(fractional_coords_w, boundaries, right=True)
+
+            pos_ids = (bucket_coords_h[:, None] * self.num_patches_per_side + bucket_coords_w).flatten()
+            position_ids[batch_idx][p_attn_mask.view(-1)] = pos_ids
+        return position_ids
+
+
+class Phi4MMConfigBehavior(str, enum.Enum):
+    AUDIO_EMBEDDINGS = "audio_embeddings"
+    AUDIO_ENCODER = "audio_encoder"
+    AUDIO_FORWARD_EMBEDDINGS = "audio_forward_embeddings"
+    AUDIO_VISION_PROJECTION = "audio_vision_projection"
+    AUDIO_SPEECH_PROJECTION = "audio_speech_projection"
+    LANGUAGE = "language"
+    TEXT_EMBEDDINGS = "text_embeddings"
+    VISION_PROJECTION = "vision_projection"
+    VISION_EMBEDDINGS = "vision_embeddings"
+
+
+@register_in_tasks_manager(
+    "phi4mm", *["image-text-to-text", "automatic-speech-recognition"], library_name="transformers"
+)
+@register_in_tasks_manager(
+    "phi4-multimodal", *["image-text-to-text", "automatic-speech-recognition"], library_name="transformers"
+)
+class Phi4MMOpenVINOConfig(BaseVLMOpenVINOConfig):
+    SUPPORTED_BEHAVIORS = [model_type.value for model_type in Phi4MMConfigBehavior]
+    NORMALIZED_CONFIG_CLASS = NormalizedVisionConfig
+    DUMMY_INPUT_GENERATOR_CLASSES = (DummyVisionInputGenerator,)
+    MIN_TRANSFORMERS_VERSION = version.parse("4.48.0")
```

Comment:
can we upgrade to 4.51 (support in transformers) 
```suggestion
    MIN_TRANSFORMERS_VERSION = version.parse("4.51.0")
```